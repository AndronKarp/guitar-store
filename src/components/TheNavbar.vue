<template>
  <b-navbar sticky type="dark" variant="info">
    <b-navbar-brand class="guitar-store-page-link" to="/"
      >Guitar Store</b-navbar-brand
    >

    <b-navbar-toggle target="nav-collapse"></b-navbar-toggle>

    <b-collapse id="nav-collapse" is-nav>
      <b-navbar-nav class="ml-auto">
        <template v-if="currentUser">
          <b-nav-item class="shopping-cart-page-link" to="/cart">
            <b-icon-cart4></b-icon-cart4>
          </b-nav-item>
          <b-dropdown :text="currentUser.displayName" variant="info">
            <b-dropdown-item @click="signOut">Sign Out</b-dropdown-item>
          </b-dropdown>
        </template>

        <template v-else>
          <b-button
            class="sign-up-button mr-2"
            to="/join"
            squared
            variant="info"
            >Sign Up</b-button
          >
          <b-button
            class="sign-in-button"
            to="/login"
            squared
            variant="outline-light"
            >Sign In</b-button
          >
        </template>
      </b-navbar-nav>
    </b-collapse>
  </b-navbar>
</template>

<script>
import { mapGetters } from "vuex";
import { auth } from "../configs/firebase";
import moveToPage from "../mixins/move-to-page";

export default {
  computed: {
    ...mapGetters(["currentUser"])
  },
  methods: {
    async signOut() {
      await auth.signOut();
      this.moveTo("PageSignIn");
      this.$store.dispatch("clearCart");
    }
  },
  mixins: [moveToPage]
};
</script>

<style></style>
