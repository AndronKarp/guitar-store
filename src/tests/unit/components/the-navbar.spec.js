import { shallowMount, createLocalVue } from "@vue/test-utils";
import TheNavbar from "@/components/TheNavbar";
import { BootstrapVue, IconsPlugin } from "bootstrap-vue";

const localVue = createLocalVue();

localVue.use(BootstrapVue);
localVue.use(IconsPlugin);

describe("TheNavbar.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(TheNavbar, {
      localVue
    });
  });

  test("renders navigation bar", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("brand link has correct path", () => {
    const brandLink = wrapper.find(".guitar-store-page-link");
    const path = brandLink.attributes("to");
    expect(path).toBe("/");
  });

  test("cart icon link has correct path", () => {
    const cartIconLink = wrapper.find(".cart-page-link");
    const path = cartIconLink.attributes("to");
    expect(path).toBe("/cart");
  });
});
