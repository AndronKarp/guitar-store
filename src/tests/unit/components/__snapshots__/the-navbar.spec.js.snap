// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheNavbar.vue renders navigation bar 1`] = `
<b-navbar-stub tag="nav" type="dark" variant="info">
  <b-navbar-brand-stub tag="div" target="_self" to="/" class="guitar-store-page-link">Guitar Store</b-navbar-brand-stub>
  <b-navbar-toggle-stub label="Toggle navigation" target="nav-collapse"></b-navbar-toggle-stub>
  <b-collapse-stub id="nav-collapse" isnav="true" tag="div">
    <b-navbar-nav-stub tag="ul">
      <b-nav-item-stub target="_self">Contact Us!</b-nav-item-stub>
    </b-navbar-nav-stub>
    <b-navbar-nav-stub tag="ul" class="ml-auto">
      <b-nav-item-stub target="_self" to="/cart" right="" class="cart-page-link d-flex align-items-center">
        <b-icon-stub icon="cart4" fontscale="1" scale="1" rotate="0" shifth="0" shiftv="0"></b-icon-stub>
      </b-nav-item-stub>
      <b-nav-item-dropdown-stub right="true" offset="0" boundary="scrollParent" text="" role="menu">
        <b-dropdown-item-stub target="_self">Profile</b-dropdown-item-stub>
        <b-dropdown-item-stub target="_self">Sign Out</b-dropdown-item-stub>
      </b-nav-item-dropdown-stub>
    </b-navbar-nav-stub>
  </b-collapse-stub>
</b-navbar-stub>
`;
